pkg/reconciler/autoscaling/hpa/keda_hpa.go

```diff
@@ -0,0 +1,160 @@
+	areconciler "knative.dev/serving/pkg/reconciler/autoscaling"
+	"knative.dev/serving/pkg/reconciler/autoscaling/config"
+
+	"knative.dev/autoscaler-keda/pkg/client/clientset/versioned"
+	kedav1alpha1 "knative.dev/autoscaler-keda/pkg/client/listers/keda/v1alpha1"
+	"knative.dev/autoscaler-keda/pkg/reconciler/autoscaling/hpa/resources"
+)
+
+// Reconciler implements the control loop for the HPA resources.
+type Reconciler struct {
+	*areconciler.Base
+
+	kubeClient kubernetes.Interface
+	kedaClient versioned.Interface
+	kedaLister kedav1alpha1.ScaledObjectLister
+	hpaLister  autoscalingv2listers.HorizontalPodAutoscalerLister
+}
+
+// Check that our Reconciler implements pareconciler.Interface
+var _ pareconciler.Interface = (*Reconciler)(nil)
+
+// ReconcileKind implements Interface.ReconcileKind.
+func (c *Reconciler) ReconcileKind(ctx context.Context, pa *autoscalingv1alpha1.PodAutoscaler) pkgreconciler.Event {
+	ctx, cancel := context.WithTimeout(ctx, pkgreconciler.DefaultTimeout)
+	defer cancel()
+
+	logger := logging.FromContext(ctx)
+	logger.Debug("PA exists")
+
+	var hpa *v2.HorizontalPodAutoscaler
+
+	dScaledObject := resources.MakeScaledObject(pa, config.FromContext(ctx).Autoscaler)
+	scaledObj, err := c.kedaLister.ScaledObjects(pa.Namespace).Get(dScaledObject.Name)
+	if errors.IsNotFound(err) {
+		logger.Infof("Creating Scaled Object %q", dScaledObject.Name)
+		if scaledObj, err = c.kedaClient.KedaV1alpha1().ScaledObjects(dScaledObject.Namespace).Create(ctx, dScaledObject, metav1.CreateOptions{}); err != nil {
+			pa.Status.MarkResourceFailedCreation("ScaledObject", dScaledObject.Name)
+			return fmt.Errorf("failed to create ScaledObject: %w", err)
+		}
+	} else if err != nil {
+		return fmt.Errorf("failed to get ScaledObject: %w", err)
+	} else if !metav1.IsControlledBy(scaledObj, pa) {
+		// Surface an error in the PodAutoscaler's status, and return an error.
+		pa.Status.MarkResourceNotOwned("ScaledObject", dScaledObject.Name)
+		return fmt.Errorf("PodAutoscaler: %q does not own ScaledObject: %q", pa.Name, dScaledObject.Name)
+	}
+	if !equality.Semantic.DeepEqual(dScaledObject.Spec, scaledObj.Spec) {
+		logger.Infof("Updating ScaledObject %q", dScaledObject.Name)
+		if _, err := c.kedaClient.KedaV1alpha1().ScaledObjects(pa.Namespace).Update(ctx, dScaledObject, metav1.UpdateOptions{}); err != nil {
+			return fmt.Errorf("failed to update ScaledObject: %w", err)
+		}
+	}
+	hpa, err = c.hpaLister.HorizontalPodAutoscalers(pa.Namespace).Get(pa.Name)
+	if errors.IsNotFound(err) {
+		return nil // skip wait to be triggered by hpa events eg. creation
+	}
+
+	if scaledObj.Spec.MinReplicaCount != nil { // if nil scaling is disabled for hpa
+		if hpa.Status.DesiredReplicas < *scaledObj.Spec.MinReplicaCount {
+			return fmt.Errorf("Hpa initializing")
+		}
+	}
+
+	// 0 num activators will work as "all".
+	sks, err := c.ReconcileSKS(ctx, pa, nv1alpha1.SKSOperationModeServe, 0 /*numActivators*/)
```

Comment:
instead of the two comments, you could use a constant/var named `allActivators` or something like that.